Libraries/Image/RCTImageView.m

```diff
@@ -144,7 +144,19 @@ - (void)reloadImage
         }
       } else {
         if (_onLoad) {
-          NSDictionary *event = @{ @"target": self.reactTag };
+          CGSize intrinsicSize = [dimensions CGSizeValue];
```

Comment:
if dimensions is nil, then exclude the "size" from the event object (otherwise include it). This means that if we download an image that actually has 0 width or height, it will be reported to the JS code.